src/main/kotlin/eu/europa/ec/eudi/sdjwt/RecreateClaims.kt

```diff
@@ -225,11 +225,12 @@ private class DiscloseObject(
     private fun discloseArray(
         currentPath: ClaimPath,
         jsonArray: JsonArray,
-    ): JsonArray =
-        jsonArray
-            .zip(0..<jsonArray.size)
-            .mapNotNull { (element, index) -> discloseArrayElement(currentPath, element, index) }
+    ): JsonArray {
```

Comment:
@dzarras 
Personally I find somehow confusing the `?.also { }` which changes a state variable outside the the lambda.
Perhaps, a more imperative implementation gives a more clear view

```kotlin
buildJsonArray {
    var index = 0
    jsonArray.forEach { element ->
        discloseArrayElement(currentPath, element, index)
            ?.let { add(it); index++ }
    }
}
```